src/auth_endpoints/signup.auth.ts

```diff
@@ -1,5 +1,44 @@
 import { Request, Response } from "express";
+import prismaClient from "../client.js";
+import crypto, { pbkdf2, pbkdf2Sync } from "crypto";
+import { validateRequestBody } from "../utils/auth/validate.request.body.js";
+import { hashPassword } from "../utils/auth/hash.password.js";
+import { isUsernameTaken } from "../utils/auth/username.availability.js";
 
 export async function signupAuth(req: Request, res: Response, next: any) {
-	console.log("muvees backend sign up endpoint called");
+	const isRequestBodyValid = await validateRequestBody(
+		req.body.username,
+		req.body.password,
+	);
+
+	if (!isRequestBodyValid) {
+		res.status(400).send("invalid username or password");
+	}
+
+	if (await isUsernameTaken(req.body.username)) {
+		res.status(400).send("username is taken by another user");
+	}
```

Comment:
This will result in a bug. If the request body is not valid, it will still continue execution, ditto for the other if below.
or does the `send` function terminate the function execution?
```suggestion
	if (!isRequestBodyValid) {
		res.status(400).send("invalid username or password");
		return;
	}

	if (await isUsernameTaken(req.body.username)) {
		res.status(400).send("username is taken by another user");
		return;
	}
```